packages/plugins/i18n/server/src/controllers/validate-locale-creation.ts

```diff
@@ -23,7 +23,7 @@ const validateLocaleCreation: Common.MiddlewareHandler = async (ctx, next) => {
     return next();
   }
 
-  const locale = get('plugins.i18n.locale', query);
```

Comment:
seeing this , how come we were using the "locale" parameter in other endpoints? If only plugins.i18n.locale was read here?

Reply:
Extracting the locale from the query within i18n only happens in PUT and POST endpoints (single-type/collection-type create, update, publish) see here https://github.com/strapi/strapi/blob/5a804de4144e88d598169a88403360fb8ef03da6/packages/plugins/i18n/server/src/register.ts#L24

I think this is how other CM endpoints were working okay. But in [fix/cm-publishing-locales](https://github.com/strapi/strapi/tree/fix/cm-publishing-locales) we have actually changed the query sent from the FE on PUT and POST e.g. https://github.com/strapi/strapi/blob/a6f23db9df5335e30072faf64ab2c554d88ab6a4/packages/core/admin/admin/src/content-manager/components/ContentTypeFormWrapper.tsx#L469

Which is why publishing behaviour was broken because the locale was in the wrong part of the query

wdyt the best way forward is? I think we could revert the format of PUT and POST queries on the FE if it's not okay to do this in v4

Reply:
oh I see! If this only affects the PUT POST actions lets move forward with this, but lets be sure we are not using those parameters somewhere else !